resolwe/flow/management/commands/list_docker_images.py

```diff
@@ -0,0 +1,64 @@
+""".. Ignore pydocstyle D400.
+
+==================
+List Docker images
+==================
+
+"""
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from resolwe.flow.models import Process
+
+
+class Command(BaseCommand):
+    """List Docker images used by processes."""
+
+    help = "List Docker images used by processes in either plain text or YAML (for Ansible)"
+
+    def add_arguments(self, parser):
+        """Add an argument to specify output format."""
+        parser.add_argument('--format', dest='format', default='plain',
+                            help="Set output format ('plain' [default] or 'yaml')")
+
+    def handle(self, *args, **options):
+        """Handle command list_docker_images."""
+        # Check that the specified output format is valid
+        if options['format'] != 'plain' and options['format'] != 'yaml':
+            raise CommandError("Unknown output format: %s" % options['format'])
+
+        # This Docker image is used if a process doesn't specify its own
+        default_docker_image = settings.FLOW_EXECUTOR['CONTAINER_IMAGE']
+
+        # Gather a list of all custom Docker images the processes are using
+        unique_docker_images = set(
+            p.requirements['executor']['docker']['image']
+            for p in Process.objects.filter(requirements__icontains='docker'))
+
+        unique_docker_images.add(default_docker_image)
+
+        # Figure out whether we should output in YAML
+        want_yaml = options['format'] == 'yaml'
+
+        # Assemble a list of Docker images
+        out = ''
+
+        for i in unique_docker_images:
```

Comment:
Use [PyYAML](http://pyyaml.org/wiki/PyYAMLDocumentation) instead of manually generating yaml.